src/sampleutil.cpp

```diff
@@ -17,14 +17,62 @@ typedef qint32 int32_t;
 // When changing, be carefull to not prevent the vectorizing
 // https://gcc.gnu.org/projects/tree-ssa/vectorization.html
 
+
+static inline bool useAlignedAlloc() {
+    // Will return true on MSVC 32 bit builds and falls for
+    // Linux 32 and 64 bit builds
+    return (sizeof(long double) == 8 && sizeof(CSAMPLE*) <= 8 &&
+            sizeof(CSAMPLE*) == sizeof(size_t));
+}
+
 // static
 CSAMPLE* SampleUtil::alloc(unsigned int size) {
```

Comment:
At some point we should consider enforcing the use of SampleUtil::free (instead of delete) with a wrapper type or something. I think a raw delete on a SampleUtil::alloc'd pointer will throw an exception with the alignment tricks. 

For example: https://code.google.com/p/vertex-vertex/source/browse/trunk/util/undeletable_ptr.hpp